contracts/Marketplace.sol

```diff
@@ -170,17 +187,25 @@ contract Marketplace is Ownable, IMarketplace {
         return true;
     }
 
-
-    function createProduct(bytes32 id, string name, address beneficiary, uint pricePerSecond, Currency currency, uint minimumSubscriptionSeconds) public whenNotHalted {
+    /*
+    createProductWhitelist is createProduct with extra whitelist arg. truffle test doesn't like 2 functions named createProduct,
+    although 2 functions with differing numbers of args is valid Solidity 
+    */
+    function createProductWhitelist(bytes32 id, string name, address beneficiary, uint pricePerSecond, Currency currency, uint minimumSubscriptionSeconds, bool requiresWhitelist) public whenNotHalted {
```

Comment:
@hpihkala  do we want to add (... bool requiresWhitelist) param to createProduct? I created a new createProductWhitelist with the added param because apparently Truffle doesn't like function overloading. Two createProduct methods with differing numbers of args are allowed by solc, but truffle complains that I'm giving createProduct the wrong number of arguments.

Reply:
We could also go with "sensible defaults", that is, doesn't require whitelisting, but after first address is added, then switches it on?

Reply:
Also note that the new marketplace contract is chained to the old one. Are we ok have 2 different APIs for Marketplace? I don't think it's a big problem. Marketplace2 could use the old createProduct (no whitelist arg) in its interface definition to interact with Marketpalce1, and then Marketplace3 (next version) could use the new createProduct (with whiteilst arg) to interact with Marketplace2.

Reply:
I don't think breaking the ABI compatibility is all that bad at this point. We aren't REALLY committed to supporting external Marketplace apps, we only have our own, and it will simply create all new products to the Marketplace2.

Reply:
I mean, at some point we have to "freeze the ABI" and never touch some IMarketplace ever again. But it's not yet I think.

Reply:
maybe it should be IMarketplace but IMarketplace1 or IFirstMarketplace?

Reply:
ok, IMarketplace1 it is

Reply:
Updating the interface does require some change management. We need to log the frontend changes needed into some JIRA ticket for the frontend guys to implement when the new marketplace contract is available (in staging/production). I made [this ticket](https://streamr.atlassian.net/browse/PLATFORM-449), please check.